homeassistant/components/deako/light.py

```diff
@@ -0,0 +1,121 @@
+"""Binary sensor platform for integration_blueprint."""
+import logging
+from typing import Any
+
+from pydeako.deako import Deako
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    add_entities: AddEntitiesCallback,
+) -> None:
+    """Configure the platform."""
+    client: Deako = hass.data[DOMAIN][config.entry_id]
+
+    devices = client.get_devices()
+    if len(devices) == 0:
+        # If deako devices are advertising on mdns, we should be able to get at least one device
+        _LOGGER.warning("No devices found from local integration")
+        await client.disconnect()
+        return
+    lights = [DeakoLightSwitch(client, uuid) for uuid in devices]
+    add_entities(lights)
+
+
+class DeakoLightSwitch(LightEntity):
+    """Deako LightEntity class."""
+
+    client: Deako
+    uuid: str
+
+    def __init__(self, client: Deako, uuid: str) -> None:
+        """Save connection reference."""
+        self.client = client
+        self.uuid = uuid
+        self.client.set_state_callback(self.uuid, self.on_update)
+
+    def on_update(self) -> None:
+        """State update callback."""
+        self.schedule_update_ha_state()
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Returns device info in HA digestable format."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.uuid)},
+            name=self.name,
+            manufacturer="Deako",
+            model="dimmer"
+            if ColorMode.BRIGHTNESS in self.supported_color_modes
+            else "smart",
+        )
+
+    @property
+    def unique_id(self) -> str:
+        """Return the ID of this Deako light."""
+        return self.uuid
+
+    @property
+    def name(self) -> str:
+        """Return the name of the Deako light."""
+        name = self.client.get_name(self.uuid)
+        return name or f"Unknown device: {self.uuid}"
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the light is on."""
+        state = self.client.get_state(self.uuid)
+        if state is not None:
+            power = state.get("power", False)
+            if isinstance(power, bool):
+                return power
+        return False
+
+    @property
+    def brightness(self) -> int:
+        """Return the brightness of this light between 0..255."""
+        state = self.client.get_state(self.uuid)
```

Comment:
Currently you are requesting the state for every property every update. I would suggest moving the whole update into `update` or `async_update` depending if the call is async or not. You can then set all the properties using shorthands.

Reply:
Good point 🤔 thanks!